front/src/applications/stdcm/hooks/useStdcmResults.ts

```diff
@@ -0,0 +1,212 @@
+
+      if (geometry && operational_points && electrifications) {
+        const pathStepsWihPosition = compact(pathSteps).map((step, i) => ({
+          ...step,
+          positionOnPath: path.path_items_positions[i],
+        }));
+
+        const suggestedOperationalPoints: SuggestedOP[] = formatSuggestedOperationalPoints(
+          operational_points,
+          geometry,
+          path.length
+        );
+
+        const updatedSuggestedOPs = upsertViasInOPs(
+          suggestedOperationalPoints,
+          pathStepsWihPosition
+        );
+
+        setPathProperties({
+          electrifications,
+          geometry,
+          suggestedOperationalPoints: updatedSuggestedOPs,
+          allVias: updatedSuggestedOPs,
+          length: path.length,
+        });
+      }
+    };
+
+    if (infraId && stdcmResponse) {
+      const { path } = stdcmResponse;
+      getPathProperties(infraId, path);
+    }
+  }, [stdcmResponse]);
+
+  useEffect(() => {
+    const getSpaceTimeChartData = async (
+      _infraId: number,
+      path: PathfindingResultSuccess,
+      trainSchedulesIds: number[],
+      stdcmSimulation: StdcmV2SuccessResponse['simulation'],
+      stdcmDepartureTime: string,
+      stdcmRollingStockLength: number
+    ) => {
+      const { blocks, routes, track_section_ranges } = path;
+      const projectPathTrainResult = await projectTrainSchedules({
+        infra: _infraId,
+        ids: trainSchedulesIds,
+        projectPathInput: { blocks, routes, track_section_ranges },
+      }).unwrap();
+
+      const newSpaceTimeData = Object.keys(projectPathTrainResult).reduce((result, trainId) => {
+        const currentProjectedTrain = projectPathTrainResult[trainId];
+
+        const matchingTrain = trainSchedules?.find((train) => train.id === +trainId);
+
+        const formattedProjectedPathTrainResult = formatSpaceTimeData(
+          trainId,
+          currentProjectedTrain,
+          matchingTrain?.train_name
+        );
+        result.push(formattedProjectedPathTrainResult);
+        return result;
+      }, [] as TrainSpaceTimeData[]);
+
+      const departureTimeInSec = convertDepartureTimeIntoSec(stdcmDepartureTime);
+      const stdcmProjectedTrain: TrainSpaceTimeData = {
+        id: STDCM_TRAIN_ID,
+        trainName: 'stdcm',
+        spaceTimeCurves: [
+          stdcmSimulation.final_output.times.map((time, index) => ({
+            // time refers to the time elapsed since departure so we need to add it to the start time
+            time: departureTimeInSec + ms2sec(time),
+            headPosition: mmToM(stdcmSimulation.final_output.positions[index]),
+            tailPosition: mmToM(
+              stdcmSimulation.final_output.positions[index] - stdcmRollingStockLength
+            ),
+          })),
+        ],
+        signal_updates: [],
+        rolling_stock_length: stdcmRollingStockLength,
+        departure_time: stdcmDepartureTime,
+      };
```

Comment:
same for this  i would create another function to test and make a more specific block function

Reply:
Done :+1: